src/main/java/com/linecorp/armeria/internal/grpc/NettyWritableBufferAllocator.java

```diff
@@ -0,0 +1,33 @@
+package com.linecorp.armeria.internal.grpc;
+
+import io.grpc.internal.WritableBuffer;
+import io.grpc.internal.WritableBufferAllocator;
+import io.netty.buffer.PooledByteBufAllocator;
+
+/**
+ * The default allocator for {@link NettyWritableBuffer}s used by the Netty transport. We set a
+ * minimum bound to avoid unnecessary re-allocation for small follow-on writes and to facilitate
+ * Netty's caching of buffer objects for small writes. We set an upper-bound to avoid allocations
+ * outside of the arena-pool which are orders of magnitude slower. The Netty transport can receive
+ * buffers of arbitrary size and will chunk them based on flow-control so there is no transport
+ * requirement for an upper bound.
+ *
+ * <p>Note: It is assumed that most applications will be using Netty's direct buffer pools for
+ * maximum performance.
+ *
+ * NOTE: Forked from grpc-java.
+ */
+public class NettyWritableBufferAllocator implements WritableBufferAllocator {
+
+    // Use 4k as our minimum buffer size.
+    private static final int MIN_BUFFER = 4096;
+
+    // Set the maximum buffer size to 1MB
+    private static final int MAX_BUFFER = 1024 * 1024;
+
+    @Override
+    public WritableBuffer allocate(int capacityHint) {
+        capacityHint = Math.min(MAX_BUFFER, Math.max(MIN_BUFFER, capacityHint));
+        return new NettyWritableBuffer(PooledByteBufAllocator.DEFAULT.buffer(capacityHint, capacityHint));
```

Comment:
No. It's safe to use the default `PooledByteBufAllocator` though because it's what Armeria uses at the moment. I'm not sure if it's a good idea to add an allocator option to `Server` and `ClientFactory` yet, so let's leave it as it is for now.